src/main/scala/monix/kafka/KafkaConsumer.scala

```diff
@@ -0,0 +1,140 @@
+package monix.kafka
+
+import com.typesafe.scalalogging.StrictLogging
+import monix.execution.Ack.Stop
+import monix.execution.FutureUtils.extensions.FutureExtensions
+import monix.execution.cancelables.SingleAssignmentCancelable
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.{Observable, Observer}
+import monix.reactive.observers.Subscriber
+import org.apache.kafka.clients.consumer.{ConsumerRecord, ConsumerRecords, KafkaConsumer => ApacheKafkaConsumer}
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
+import scala.collection.JavaConverters._
+
+/** An `Observable` implementation that reads messages from `Kafka`.
+  *
+  * You need to call `addTopicSubscription` in order for the Observable to start sending messages from that topic.
+  */
+final class KafkaConsumer[K, V] private
+  (config: KafkaConsumerConfig, ioScheduler: Scheduler)
```

Comment:
I would take a `ApacheKafkaConsumer` because the user might want to do some special initialization. In our project, if you remember, we started doing `seekToEnd()` on that consumer, prior to consuming from it.

And then you can have 2 builders, one that takes an `ApacheKafkaConsumer` and one that takes a `KafkaConsumerConfig`.

Also, the `ApacheKafkaConsumer` input parameter should probably be a `Task[ApacheKafkaConsumer]`, because we want to repeat its creation whenever we subscribe.